lib/puppet.rb

```diff
@@ -42,11 +42,10 @@ module Puppet
   require 'puppet/environments'
 
   class << self
-    gettext_config_file = Puppet::GettextConfig.puppet_locale_path
-    unless Puppet::GettextConfig.initialize(gettext_config_file, Puppet::GettextConfig.translation_mode(gettext_config_file))
-      # Stub out gettext's `_` and `n_()` methods, which attempt to load translations,
-      # with versions that do nothing
-      require 'puppet/gettext/stubs'
+    Puppet::GettextConfig.create_text_domain('production')
+    locale_dir = Puppet::GettextConfig.puppet_locale_path
+    if Puppet::GettextConfig.load_translations('puppet', locale_dir, Puppet::GettextConfig.translation_mode(locale_dir))
+      Puppet::GettextConfig.set_locale(Locale.current.language)
```

Comment:
Shouldn't we always set_locale? Modules might need it even if loading puppet translations fails. But I also don't see why loading puppet translations would fail.

Reply:
And set_locale doesn't guard in-case fast_gettext didn't load. So I guess this is a decent option.